mlflow/sklearn/__init__.py

```diff
@@ -455,6 +455,66 @@ def load_model(model_uri):
     )
 
 
+from collections import defaultdict
+
+
+class _AutologTrainingStatus:
+    def __init__(self, last_mlflow_run_id=None, model_id=None):
+        self.last_mlflow_run_id = last_mlflow_run_id
+        self.model_id = model_id
+        self.log_test_metric_count_map = defaultdict(lambda: 0)
+        self.predict_result_instance_ids = []
+        self.in_fit_call_scope = False
+        self.in_scorer_call_scope = False
```

Comment:
To achieve “autolog evaluation metric”, We need 3 patch point:
(1) patch `fit` series methods
(2) patch `predict` series methods
(3) patch `metric computation` series methods

We do “autolog evaluation metric” in patch point (3), but, we need ensure:
- patch point (3) will use the same “mlflow run” with the run used in (1)
- patch point (3) will compute metric on a prediction result dataset, the prediction result dataset must be generated by the model used in (1) fit call.

The global data structure is designed for achieving the above 2 goals.